backend/internal/handlers/board_handler.go

```diff
@@ -40,7 +84,51 @@ func CreateBoard(c *gin.Context) {
 		return
 	}
 
-	// Return response
+	// Get the owner role
+	var ownerRole models.Role
+	if err := database.DB.Where("name = ?", "owner").First(&ownerRole).Error; err != nil {
+		// If role doesn't exist, still return the board but log the warning
+		c.JSON(http.StatusCreated, gin.H{
+			"board": BoardResponse{
+				ID:              board.ID,
+				Title:           board.Title,
+				Description:     board.Description,
+				BackgroundColor: board.BackgroundColor,
+				OwnerID:         board.OwnerID,
+				CreatedAt:       board.CreatedAt,
+				UpdatedAt:       board.UpdatedAt,
+			},
+			"warning": "Board created but owner role not assigned. Please run migrations.",
+		})
+		return
```

Comment:
- [ ] Treat this as a critical error and return an internal server error (500 HTTP status code). A missing role due to lack of migration is critical error which should terminate the HTTP request and drop the attention of the developers.
- [ ] Wrap all the database queries (whether fetch or insert or update operations) in this handler in a single transaction. I think the `gorm` (I guess that's the ORM you are using) should provide a way of doing this. Wrapping with a transaction ensures that during this board creation operation, if any error is encountered from any database or non-database operation, any incomplete inserts or updates will be rolled back and avoid orphaned resources. Same applies to another handlers.